package.json

```diff
@@ -105,13 +105,17 @@
 		"rules": {
 			"no-alert": "off",
 			"import/no-unassigned-import": "off",
-			"import/no-unresolved": "warn"
+			"import/no-unresolved": "warn",
+			"import/named": "warn"
 		},
 		"overrides": [
 			{
 				"files": "**/*.+(ts|tsx)",
 				"extends": "xo-typescript",
 				"rules": {
+					"@typescript-eslint/no-misused-promises": "off",
+					"@typescript-eslint/require-await": "off",
+					"@typescript-eslint/no-explicit-any": "off",
```

Comment:
Good in theory, but makes us have to type on-off things just for the heck of it:

- Code examples:


	```ts
	(window as any).select = select;
	```


	```ts
	const editor: CodeMirrorInstance = document.querySelector<any>('.CodeMirror').CodeMirror;
	```

	```ts
	(global as any).navigator = window.navigator;
	(global as any).document = window.document;
	(global as any).location = new URL('https://github.com');
	// Which becomes at least, with no advantage
	(global as unknown as typeof window).navigator = window.navigator;
	(global as unknown as typeof window).document = window.document;
	(global as unknown as typeof window).location = new URL('https://github.com');
	```

- It'd force us to type API calls
- Also kills `AnyObject` which is pretty much required for the API handler

Reply:
The idea was that you explicitly disable the rule where you want `any` to make it inconvenient enough to use that it's not misused, but I encountered some annoyance with it too, so I'll disable it for now: https://github.com/xojs/eslint-config-xo-typescript/commit/bbbf88972520e8af3439ba2ab3f1a1c40e68041c

Reply:
> The idea was that you explicitly disable the rule where you want `any` to make it inconvenient enough to use that it's not misused

Maybe that is worth it, I missed this one just recently: https://github.com/sindresorhus/refined-github/pull/2393/files#diff-f8fed5d366f31fce028a9c8ceca38c3eR6

The lines in my examples would just have to use `eslint-disable-next-line` or be wrapped by `disabled/enable`

Reply:
I think it will be easier to enable in a year or so when TS has matured more and we need less `any` in general.

Reply:
What other problems have you had?

Do you think there will ever be a solution to my examples? They basically are:

- setting a global (without having to create a property on Global/Window)
- setting/getting one-off properties. It could probably already be fixed with:
	```ts
	type CodeMirrorElement = CodeMirrorInstance & {CodeMirror: CodeMirrorInstance}
	const editor = document.querySelector<CodeMirrorElement>('.CodeMirror').CodeMirror;
	```
- we just want to avoid typing of certain parts like APIs because it's extra work that **has to** be updated when the query changes, otherwise the API response won't actually match the types (in essence, manually typing the API is as bad as an `as`sertion). Also the API response ALWAYS has to start from `AnyObject`, there's no workaround for that.

Reply:
I had problems with some bad types from Electron and React. Basically out of my control. I also had to use `any` for some things like this: https://github.com/sindresorhus/caprine/blob/7965b36c11ae1f7ccb2185bebdc85e9fc3d01078/source/menu.ts#L92